build.assets/Makefile

```diff
@@ -201,7 +201,7 @@ test: buildbox
 		$(DOCKERFLAGS) $(BCCFLAGS) $(NOROOT) -t $(BUILDBOX) \
 		/bin/bash -c \
 		"examples/etcd/start-etcd.sh & sleep 1; \
-		gcloud beta emulators firestore start --host-port=localhost:8618 & sleep 1; \
+		gcloud beta emulators firestore start --host-port=localhost:8618 & sleep 1 || exit 1; \
```

Comment:
This || will chain off the sleep, rendering it meaningless:

```console
$ gcloud beta emulators firestore start --host-port=localhost:8618 & sleep 1 || echo hey
[1] 379596
ERROR: (gcloud.beta.emulators.firestore.start) To use the Google Cloud Firestore emulator, a Java 8+ JRE must be installed and on your system PATH
[1]  + exit 1     gcloud beta emulators firestore start --host-port=localhost:8618
```

I'd suggest a query, then a start in the background:

```console
$ gcloud beta emulators firestore start --help > /dev/null
$ cloud beta emulators firestore start --host-port=localhost:8618 &
$ sleep 1
...
```

Reply:
How about this?

```bash
type gcloud 2>&1 >/dev/null || exit 1; \
gcloud -q beta emulators firestore start --host-port=localhost:8618 & sleep 1; \
```

It'll error out if `gcloud` isn't installed, and the `-q` means that if the `beta` command set isn't installed for gcloud, it'll also install it (this shouldn't ever happen, but just in case). I can't find a way to make `gcloud` exit if it's not installed rather than auto-installing.

Reply:
LGTM.